staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -273,20 +296,50 @@ func (s *SpdyRoundTripper) tlsConn(ctx context.Context, rwc net.Conn, targetHost
 // dialWithoutProxy dials the host specified by url, using TLS if appropriate.
 func (s *SpdyRoundTripper) dialWithoutProxy(ctx context.Context, url *url.URL) (net.Conn, error) {
 	dialAddr := netutil.CanonicalAddr(url)
-	dialer := s.Dialer
-	if dialer == nil {
-		dialer = &net.Dialer{}
+	dial := s.DialContext
+	// prefer DialContext
+	if dial == nil {
+		// but fallback to the provided Dialer and then a generic net.Dialer
+		if s.Dialer != nil {
+			dial = s.Dialer.DialContext
+		} else {
+			nd := &net.Dialer{}
+			dial = nd.DialContext
+		}
 	}
 
+	conn, err := dial(ctx, "tcp", dialAddr)
+	if err != nil {
+		return nil, err
+	}
 	if url.Scheme == "http" {
```

Comment:
actually, this becomes the same (just adapted to this part) as

```
	if req.URL.Scheme == "https" {
		return s.tlsConn(req.Context(), proxyDialConn, targetHost)
	}
	return proxyDialConn, nil
```

we are already using s.tlsConn @liggitt 🤷

Reply:
we have to make sure that our usage of s.tlsConn never drifts from `tls.Dialer{}.DialContext`  🤔